firedrake/checkpointing.py

```diff
@@ -716,6 +720,124 @@ def _save_mesh_topology(self, tmesh):
+
+        Parameters
+        ----------
+        mesh : firedrake.mesh.MeshGeometry
+            The mesh under which the function should be searched.
+        name : str
+            The name of the function for which indices, times, and timesteps are to be stored. This
+            function must be a part of the provided mesh.
+        idx : int
+            The index of the function `name` in the timestepping sequence. It uniquely identifies the
+            function's position in the sequence of timesteps.
+        t : float, optional
+            The time associated with the function `name` at the given index `idx`.
+            If not provided, a default real value is used.
+        timestep : float, optional
+            The timestep value associated with the function `name` at the given index `idx`.
+            If not provided, a default real value is used.
+
+        Raises
+        ------
+        ValueError
+            If `t` or `timestep` are provided but cannot be converted to float.
+
+        Notes
+        -----
+        The method identifies the correct function (if mixed) based on the provided `name` and stores
+        the `idx`, `t`, and `timestep` attributes accordingly. It concatenates these values with any
+        existing indices, times, and timesteps for the function, ensuring a comprehensive record of
+        all timesteps.
+        """
+
+        t = CheckpointFile.DEFAULT_REAL if t is None else t
+        timestep = CheckpointFile.DEFAULT_REAL if timestep is None else timestep
+        try:
+            t = float(t)
+            timestep = float(timestep)
+        except ValueError:
+            raise ValueError(
+                """t and timestep must be convertible to float."""
+            )
+
+        # check if the function is mixed
+        if name in self._get_mixed_function_name_mixed_function_space_name_map(mesh.name):
+            V_name = self._get_mixed_function_name_mixed_function_space_name_map(mesh.name)[name]
+            path = self._path_to_mixed_function(mesh.name, V_name, name)
+        elif name in self._get_function_name_function_space_name_map(self._get_mesh_name_topology_name_map()[mesh.name], mesh.name):
+            tmesh_name = self._get_mesh_name_topology_name_map()[mesh.name]
+            V_name = self._get_function_name_function_space_name_map(tmesh_name, mesh.name)[name]
+            path = self._path_to_function(tmesh_name, mesh.name, V_name, name)
+
+        old_indices, old_ts, old_timesteps = self.get_timesteps(mesh, name)
+        indices = np.concatenate((old_indices, [idx]))
+        ts = np.concatenate((old_ts, [t]))
+        timesteps = np.concatenate((old_timesteps, [timestep]))
+
+        # store all timesteps and indices
+        self.set_attr(path, PREFIX_TIMESTEPPING + "_indices", indices)
+        self.set_attr(path, PREFIX_TIMESTEPPING + "_times", ts)
+        self.set_attr(path, PREFIX_TIMESTEPPING + "_timesteps", timesteps)
+
```

Comment:
```suggestion
```
Sorry, I asked you to rewrite the docstring, but I think the contents of this function can be included in `save_function` and `save_function_topology`.